pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -3159,6 +3353,23 @@ func toStatus(lb *elb.LoadBalancerDescription) *v1.LoadBalancerStatus {
 	return status
 }
 
+func v2toStatus(lb *elbv2.LoadBalancer) *v1.LoadBalancerStatus {
+	status := &v1.LoadBalancerStatus{}
+	if lb == nil {
+		glog.Error("[BUG] v2toStatus got nil input, this is a Kubernetes bug, please report")
+		return status
+	}
+
+	if aws.StringValue(lb.DNSName) != "" && (aws.StringValue(lb.State.Code) == elbv2.LoadBalancerStateEnumActive ||
+		aws.StringValue(lb.State.Code) == elbv2.LoadBalancerStateEnumProvisioning) {
```

Comment:
I'm guessing we treat provisioning as ready because otherwise we'd have to poll the loadbalancer to wait for it to be ready?  If that's the case, I agree, but can we add a comment as to that logic (or the correct reason if it's something else!)

Reply:
Sure. The other states are "active_impaired" and "failed", and in those cases, I didn't want to return a `*v1.LoadBalancerStatus` for a non-working load balancer. I added this per @andrewtandoc's above comment